src/arrays.js

```diff
@@ -1,61 +1,133 @@
 const getNthElement = (index, array) => {
-  // your code here
+  if (index >= array.length){
+  return array[index - array.length];
+  } else{
+  return array[index];
+  }
 };
 
 const arrayToCSVString = array => {
-  // your code here
+  return array.toString();
 };
 
 const csvStringToArray = string => {
-  // your code here
+  return string.split(',');
 };
 
 const addToArray = (element, array) => {
-  // your code here
+  array.push(element);
 };
 
 const addToArray2 = (element, array) => {
-  // your code here
+  const array2 = array.concat(element);
+  return array2;
 };
 
 const removeNthElement = (index, array) => {
-  // your code here
+  return array.splice(index, 1);
 };
 
 const numbersToStrings = numbers => {
-  // your code here
+  let strings = [];
+  for (let i = 0; i < numbers.length; i += 1){
+    strings.push(numbers[i].toString());
+  }
+  return strings;
 };
 
 const uppercaseWordsInArray = strings => {
-  // your code here
+  let uppercaseArray = [];
+  for (let i = 0; i < strings.length; i += 1){
+    uppercaseArray.push(strings[i].toUpperCase());
+  }
+  return uppercaseArray;
 };
 
 const reverseWordsInArray = strings => {
-  // your code here
+  /*A very difficult problem to solve for me, after some googling, I found the following:
+  https://stackoverflow.com/questions/59621299/javascript-reversing-every-string-of-an-array
+  Solution below is a simplified version of what the person asking the question got to in the end */
+  let reversedArray = [];
+  //First for loop loops through the array and processes each string
+  for (let i = 0; i < strings.length; i += 1){
+    /*Second for loop stores the letters in reverse, starting with the last character and moving down.
+    j has to be 1 less than the word's length and decreases by 1 each time, and the loop stops it reaches character 0 */
+    let reversedWord = '';
+    for (let j = strings[i].length - 1; j >= 0; j -= 1){
+      reversedWord += strings[i][j];
+    }
+    reversedArray.push(reversedWord);
+  }
+  return reversedArray;
 };
 
 const onlyEven = numbers => {
-  // your code here
+  let evenNumbers = [];
+  for (let i = 0; i < numbers.length; i += 1){
+    /*An even number is any number that gives no remainder when divided by two
+    Only numbers meeting that condition are pushed into the new array */
+    if (numbers[i] % 2 === 0){
```

Comment:
This is great stuff. Do you think you can refactor it to use an iterator method we learned on Wednesday?